### PR TITLE
CEDS-1511 Fix Summary Page Back Link

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/views/declaration/summary/summary_page.scala.html [hunk 1]
```diff
@@ -32,20 +32,22 @@
...
-                    components.back_link(controllers.declaration.routes.TransportDetailsController.displayPage(mode), mode)
-            }
-        }
+  @mode match {
+      case Mode.Draft => {
+          @components.back_link(controllers.routes.SavedDeclarationsController.displayDeclarations(), mode)
+      }
+      case Mode.Amend => {
+          @components.back_link(controllers.routes.SavedDeclarationsController.displayDeclarations(), mode)
+      }
```
Comment (SteveSugden) [inline app/views/declaration/summary/summary_page.scala.html]: Aren't those two cases doing the same thing?

File: test/unit/tools/Stubs.scala [hunk 2]
```diff
@@ -64,6 +64,7 @@ trait Stubs {
       |metrics.showSamples=false
       |metrics.jvm=false
       |metrics.logback=false
+      |draft.timeToLive=1d
```
Comment (SteveSugden) [inline test/unit/tools/Stubs.scala]: What;s this change for?

File: app/views/declaration/summary/summary_page.scala.html (hunk 1 above)
Comment (ed0906) [reply app/views/declaration/summary/summary_page.scala.html]: Good spot, think that was a copy and paste error

File: test/unit/tools/Stubs.scala (hunk 2 above)
Comment (ed0906) [reply test/unit/tools/Stubs.scala]: To render the summary page for a draft we need that adding into the config apparently,
otherwise it breaks

**Comment by ed0906** [comment]
Turns out we cant fix the back link behaviour for Amends until we change the Amend Logic from "copy the rejected dec, then edit" to "edit the rejected dec" as we would need to know the ID of the rejected declaration to be able to go back to it...

**Comment by SteveSugden** [comment]
So does this PR still need to be merged?  I'm struggling to see any change in the view logic

**Comment by ed0906** [comment]
> So does this PR still need to be merged? I'm struggling to see any change in the view logic

Well, there was some missing tests that exist now so it's probably still worth merging

**Pull request merged**